File: src/main/java/com/hoi4utils/clausewitz/exceptions/LocalizationExistsException.java
package com.hoi4utils.clausewitz.exceptions;

import com.hoi4utils.clausewitz.localization.Localization;

public class LocalizationExistsException extends Throwable {
    public LocalizationExistsException(Localization localization) {
        super("Localization already exists: " + localization.ID());
    }

    public LocalizationExistsException(Localization prevLocalization, Localization localization) {
        super("Localization already exists: " + prevLocalization + "\n\tand cannot be replaced by: " + localization);
    }

    public LocalizationExistsException(Localization prevLocalization, String text) {
        super("Localization already exists: " + prevLocalization + "\n\tand cannot be replaced with the text: " + text);
    }
}

File: src/main/scala/com/hoi4utils/exceptions/LocalizationExistsException.scala
package com.hoi4utils.exceptions

import com.hoi4utils.localization.Localization

class LocalizationExistsException(message: String) extends Exception(message) {
  def this(localization: Localization) =
    this(s"Localization already exists: ${localization.ID}")

  def this(prev: Localization, next: Localization) =
    this(
      s"Localization already exists: $prev\n\tand cannot be replaced by: $next"
    )

  def this(prev: Localization, text: String) =
    this(
      s"Localization already exists: $prev\n\tand cannot be replaced with the text: $text"
    )
}
